docs/source/extending.rst

```diff
@@ -0,0 +1,144 @@
+.. _extending:
+
+======================
+Extending the Notebook
```

Comment:
I had in mind that this section would include subsections on Notebook/Server Extensions as well.  So the hierarchy would be something like:
```
* Extending the Server
** The Contents API
** NBExtensions
** Server Extensions
** ??
...

Is there another place where docs for the other extensions already live?
```